cli/popper/commands/cmd_archive.py

```diff
@@ -115,27 +116,48 @@ def upload_snapshot(service_url, params, filename):
             "{} and the file id is {}.".format(deposition_id, file_id)
         )
     else:
-        pu.fail(
-            "Status {}: Failed to upload your snapshot. Please "
-            "try again.".format(r.status_code)
-        )
+        pu.fail("Failed to upload your snapshot. Please try again.")
 
     return deposition_id
 
 
 def add_metadata(metadata_url, params):
     """Receives the metadata_url and other parameters, reads the metadata and
     adds that to the deposit. Returns the publish_url."""
-    data = {
-        'metadata': {
-            'title': "This is the title",
-            'upload_type': 'software',
-            'description': "A suitable description",
-            'creators': [
-                {'name': 'Doe, John', 'affiliation': 'popper'},
-            ]
-        }
-    }
+
+    # Get the metadata from the .popper.yml file
+    data = pu.read_config()['metadata']
+
+    try:
+        # Change abstract to description
+        data['description'] = '<p>' + data['abstract'] + '</p>'
+        del data['abstract']
+
+        # Collect the authors
+        creators = []
+        for key in list(data.keys()):
+            if 'author' in key:
+                name, email, affiliation = map(
+                    lambda x: x.strip(), data[key].split(',')
+                )
+                if len(name.split()) == 2:
+                    name = ', '.join(name.split()[::-1])
+                creators.append({'name': name, 'affiliation': affiliation})
+                del data[key]
+        data['creators'] = creators
+
+        # Change the keywords to a list from string of comma separated values
+        data['keywords'] = list(
+            map(lambda x: x.strip(), data['keywords'].split(','))
+        )
+
+        data = {'metadata': data}
+    except KeyError:
```

Comment:
is the metadata absolutely required to upload to zenodo? If not, can we make this step (reading metadata) optional, i.e. ignore the lack of keys in the metadata dictionary and keep going?